controller/konnect/reconciler_credential_secrets.go

```diff
@@ -0,0 +1,418 @@
+package konnect
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/kong/gateway-operator/controller/konnect/constraints"
+	"github.com/kong/gateway-operator/controller/pkg/log"
+	operatorerrors "github.com/kong/gateway-operator/internal/errors"
+	k8sreduce "github.com/kong/gateway-operator/pkg/utils/kubernetes/reduce"
+
+	configurationv1 "github.com/kong/kubernetes-configuration/api/configuration/v1"
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+)
+
+const (
+	// KongCredentialTypeBasicAuth is the type of basic-auth credential, it's used
+	// as the value for konghq.com/credential label.
+	KongCredentialTypeBasicAuth = "basic-auth"
+)
+
+// KongCredentialSecretReconciler reconciles a KongPlugin object.
+type KongCredentialSecretReconciler struct {
+	developmentMode bool
+	client          client.Client
+	scheme          *runtime.Scheme
+}
+
+// NewKongCredentialSecretReconciler creates a new KongCredentialSecretReconciler.
+func NewKongCredentialSecretReconciler(
+	developmentMode bool,
+	client client.Client,
+	scheme *runtime.Scheme,
+) *KongCredentialSecretReconciler {
+	return &KongCredentialSecretReconciler{
+		developmentMode: developmentMode,
+		client:          client,
+		scheme:          scheme,
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *KongCredentialSecretReconciler) SetupWithManager(_ context.Context, mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("KongCredentialSecret").
+		For(
```

Comment:
IHAC watching secrets will cause large memory consumption like what has happened in KIC. How about adding restraints to limit the secrets to have certain labels to reduce the potential memory consumption? Also I see there is a check on labels of `Secret` in the `Reconcile`, I think we can add the label filtering in the watcher.

Reply:
Good point, I'll address that

Reply:
Have we added the predicate to filter the watched secret?